Dockerfiles/pcluster-amazonlinux-2/Dockerfile

```diff
@@ -129,6 +129,19 @@ RUN mkdir -p $(dirname "${SPACK_ROOT}") \
     && spack buildcache create -a -u /bootstrap-gcc-cache $(spack find --format '/{hash}') \
     && rm -rf $(dirname "${SPACK_ROOT}") /root/.spack
 
+ARG KEYNAME=bootstrap-gcc
+# Sign the buildcache
+RUN mkdir -p $(dirname "${SPACK_ROOT}") \
+    && git clone https://github.com/spack/spack "${SPACK_ROOT}" \
+    && pushd "${SPACK_ROOT}" && git checkout ${SPACK_COMMIT} && popd \
+    && . "${SPACK_ROOT}/share/spack/setup-env.sh" \
+    && mkdir -p /bootstrap-gcc-cache/build_cache/_pgp \
+    && spack gpg create --export /bootstrap-gcc-cache/build_cache/_pgp/${KEYNAME}.pub ${KEYNAME} stesachs@amazon.com \
```

Comment:
Isn't this leaving the private key in the container :thinking: ?

Reply:
Also, signing can be combined with buildcache creation if the key is registered before the buildcache.

Reply:
Yes, it is keeping the private key in there for now. Is there a place in GH where I can store the private key to re-use when re-running?

I know I can sign when creating the buildcache, but I wanted to leeave the layer intact as it takes a really long time ot build.

Reply:
Disclaimer: I'm not a security guy. But why signing with a private key that anybody can download?

Reply:
No, I meant somewhere private. Where only the containers built here can access. I am not at all fluid in buiuld pipelines.

Reply:
Ah, ok - so the PR is not ready yet, right?

Reply:
Well it is incremental progress. In order to upload this cache I still need ot find a permanent place for the private key.